lib/datastore.js

```diff
@@ -19,38 +19,196 @@ function MongoDataStore(config){
     if (!config.cache.cacheId) config.cache.cacheId = config.name;
   }
 
-  Object.defineProperty(this, 'config', {value:config});
+  if (!config.opts) config.opts = {};
+
+  //enforce an immediately returned error when we cannot access the db
+  config.opts.bufferMaxEntries = 0;
```

Comment:
In future we might consider a queue with a timeout to try and set once the connection is back up.

Reply:
Do we need to set this? Can't we keep the queue while the system is reconnecting?